app/controllers/api/v0/articles_controller.rb

```diff
@@ -47,7 +47,14 @@ def update
       def me
         per_page = (params[:per_page] || 30).to_i
         num = [per_page, 1000].min
-        @articles = @user.articles.
+
+        @articles = @user.articles.published # defaults to only published articles
+
+        @articles = @user.articles.published if params[:status] == "published"
```

Comment:
I think `case` would look better here (easier to understand), what do you think?

Reply:
I've never been a big fan of case/switch but you might be right:

```ruby
@articles = case params[:status]
            when "published"
              @user.articles.published
            when "unpublished"
              @user.articles.unpublished
            when "all"
              @user.articles
            else
              @user.articles.published
            end
```

I'm good either way BTW

Reply:
Neither am I! (but I won't ask to refactor this to a query object or something at the moment :D)
For me, assigning articles to `@user.articles.published` and assigning it to the same thing in some cases on the next line looks a bit weird.